backend/src/logging.rs

```diff
@@ -0,0 +1,80 @@
+use crate::app_env;
+use opentelemetry::trace::TracerProvider;
+use opentelemetry::KeyValue;
+use opentelemetry_otlp::{MetricExporter, SpanExporter, WithExportConfig};
+use opentelemetry_sdk::metrics::{PeriodicReader, SdkMeterProvider};
+use opentelemetry_sdk::trace::Tracer;
+use opentelemetry_sdk::{runtime, Resource};
+use tracing::level_filters::LevelFilter;
+use tracing_opentelemetry::{MetricsLayer, OpenTelemetryLayer};
+use tracing_subscriber::{prelude::*, registry, EnvFilter};
+
+/// The name of the service as it should appear in OpenTelemetry collectors
+const SERVICE_NAME: &str = "genconcal_backend";
+
+pub struct OtelExporters {
+    pub tracer: Tracer,
+    pub meter: SdkMeterProvider,
+}
+
+pub fn init_exporters(otlp_traces_endpoint: &str, otlp_metrics_endpoint: &str) -> OtelExporters {
+    let span_export = SpanExporter::builder()
+        .with_tonic()
+        .with_endpoint(otlp_traces_endpoint)
+        .build()
+        .expect("failed to build span exporter");
+    let meter_export = MetricExporter::builder()
+        .with_tonic()
+        .with_endpoint(otlp_metrics_endpoint)
+        .build()
+        .expect("failed to build meter exporter");
+
+    let metrics_reader = PeriodicReader::builder(meter_export, runtime::Tokio).build();
+
+    let trace_provider = opentelemetry_sdk::trace::TracerProvider::builder()
+        .with_batch_exporter(span_export, runtime::Tokio)
+        .with_resource(Resource::new([KeyValue::new("service.name", SERVICE_NAME)]))
+        .build()
+        .tracer(SERVICE_NAME);
+    let meter_provider = SdkMeterProvider::builder()
+        .with_reader(metrics_reader)
+        .with_resource(Resource::new([KeyValue::new("service.name", SERVICE_NAME)]))
+        .build();
+
+    OtelExporters {
+        tracer: trace_provider,
+        meter: meter_provider,
+    }
+}
+
+pub fn init_env_filter() -> EnvFilter {
+    EnvFilter::builder()
+        .with_default_directive(LevelFilter::INFO.into())
```

Comment:
Rgeression: Need to add a directive to EnvFilter that sets SQLX logs to "warn" by default

Reply:
Can't do this with the library unfortunately.